src/varname.jl

```diff
@@ -26,29 +29,65 @@ julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
-struct VarName{sym, T<:Tuple}
+struct VarName{sym,T<:Lens}
     indexing::T
 
-    VarName{sym}(indexing::Tuple=()) where {sym} = new{sym,typeof(indexing)}(indexing)
+    function VarName{sym}(indexing=IdentityLens()) where {sym}
+        # TODO: Should we completely disallow or just `@warn`?
+        # TODO: Does this affect performance?
```

Comment:
From DPPL benchmarks it doesn't seem to matter (this should be compiled away anways).